test/unit/mem/copy/src/BufSlicing.cpp

```diff
@@ -0,0 +1,459 @@
+            offsets
+        );
+        // Copy the subView into a new buffer
+        alpaka::mem::view::copy(
+            devQueue,
+            slicedBuffer,
+            subView,
+            subViewExtents
+        );
+        return slicedBuffer;
+    }
+
+
+    auto compareBuffer(
+        BufHost const & bufferA,
+        std::vector< Data > const & vec,
+        Vec extents
+    ) const
+    {
+        // create default buffer
+
+        Data const * const ptrA = alpaka::mem::view::getPtrNative( bufferA );
+        // Test if the buffer was not modified with the copy-operations
+        for( Idx i( 0 ); i < extents.prod( ); ++i )
+        {
+            // float and double caparison should be safe too,
+            // because the data was not modified.
+            REQUIRE( ptrA[i] == Approx( vec[static_cast<size_t>(i)] ) );
+        }
+    }
+};
+
+template<
+    typename Dim,
+    typename Idx
+>
+struct DimHelper;
+
+// 1D explicit specialisation
+template<
+    typename Idx
+>
+struct DimHelper<
+    alpaka::dim::DimInt< 1u >,
+    Idx
+>
+{
+    using Vec = alpaka::vec::Vec<
+        alpaka::dim::DimInt< 1u >,
+        Idx
+    >;
+
+
+    template< typename Data >
+    static auto cmpCorrectlySlicedBuffer(
+        Vec extents,
+        Vec extentsSubView,
+        Vec offsets
+    ) -> std::vector< Data >
+    {
+        std::vector< Data >
+            resSliced =
+            std::vector< Data >( static_cast<size_t>(extentsSubView.prod( )) );
+        alpaka::ignore_unused( extents );
+        for( Idx i = 0; i < extentsSubView[0]; ++i )
+        {
+            resSliced[static_cast<size_t>(i)] =
+                static_cast<Data>(i + offsets[0]);
+        }
+        return resSliced;
+    }
+};
+
+
+// 2D explicit specialisation
+template<
+    typename Idx
+>
+struct DimHelper<
+    alpaka::dim::DimInt< 2u >,
+    Idx
+>
+{
+    using Vec = alpaka::vec::Vec<
+        alpaka::dim::DimInt< 2u >,
+        Idx
+    >;
+
+
+    template< typename Data >
+    static auto cmpCorrectlySlicedBuffer(
+        Vec extents,
+        Vec extentsSubView,
+        Vec offsets
+    ) -> std::vector< Data >
+    {
+        std::vector< Data >
+            resSliced =
+            std::vector< Data >( static_cast<size_t>( extentsSubView.prod( ) ) );
+        for( Idx x = 0; x < extentsSubView[0]; ++x )
+        {
+            for( Idx y = 0; y < extentsSubView[1]; ++y )
+            {
+                resSliced[static_cast<size_t>(x * extentsSubView[1] + y)] =
+                    static_cast<Data>(
+                        ( x + offsets[0] ) * extents[1] + offsets[1] + y);
+            }
+        }
+        return resSliced;
+    }
+};
+
+// 3D explicit specialisation
+template< typename Idx >
+struct DimHelper<
+    alpaka::dim::DimInt< 3u >,
+    Idx
+>
+{
+    using Vec = alpaka::vec::Vec<
+        alpaka::dim::DimInt< 3u >,
+        Idx
+    >;
+
+
+    template< typename Data >
+    static auto cmpCorrectlySlicedBuffer(
```

Comment:
This method does not compare anything. It only writes the data into a 1D buffer without without padding/offsets.
Please use a different name for this.

Furthermore, the original test uses similar code to access the values. It is written in a dimension independent way:
https://github.com/alpaka-group/alpaka/blob/develop/test/common/include/alpaka/test/mem/view/Iterator.hpp#L135-L153
Would be nice if we could apply a similar logic here.

Reply:
I am not quite sure if my [implementation](https://github.com/alpaka-group/alpaka/blob/5726e8ae9ac25fcc7a8f8761ae8f7866061041a2/test/unit/mem/copy/src/BufSlicing.cpp#L308-L321) fulfills your request.

It seems that the slicing doesn't work properly with 4 dimensions.
The problem seems independent of `Idx` / `Acc` or `Datatype` and occurs in the slicing process with `slicedBuffer`.

(_cmp was short for compute, but I agree that's misleading._=